Public/Src/FrontEnd/MsBuild/PipConstructor.cs

```diff
@@ -613,6 +632,29 @@ private bool TryAddMsBuildArguments(ProjectWithPredictions project, PipDataBuild
             // Common arguments to all MsBuildExe invocations
             pipDataBuilder.AddRange(s_commonArgumentsToMsBuildExe.Select(argument => PipDataAtom.FromString(argument)));
 
+            // If process breakaway is supported and shared compilation is configured to run, set up the logger that will mimic all the proper accesses
+            // Otherwise, disable shared compilation
+            if (UseSharedCompilation)
+            {
+                var vbcsCompilerLoggerPath = AbsolutePath.Create(PathTable, Assembly.GetAssembly(typeof(PipConstructor)).Location)
+                    .GetParent(PathTable)
+                    // Depending on the framework of the MSBuild we are using, we should provide the corresponding logger
+                    // Keep in sync with BuildXL deployment
+                    .Combine(PathTable, m_resolverSettings.ShouldRunDotNetCoreMSBuild()? "dotnetcore" : "net472")

```

Comment:
Would it make sense to layout the logger dll right next to this dll so you don't have to guess the pass?

Reply:
But the decision is not made based on the location of this dll, but on whether we should run netcore or full framework MSBuild. And each version needs to live in its own folder, otherwise there might be some double-deployment problems when dlls clash because of tfm disagreement

---
In reply to: [346070749](https://github.com/microsoft/BuildXL/pull/1207#discussion_r346070749) [](ancestors = 346070749)

Reply:
Then wouldn't it be better if the dsc files did the output directory checks and copy things rather than having the c# code do it? But it's more of a nit really, doesn't matter that much :)

Reply:
But dsc files take care of the deployment of bxl itself. And therefore they need to consider both cases, since we need both versions of the logger (full and netcore). The decision of which one to use happens at runtime, for a particular bxl invocation. dsc files are not involved at runtime.

---
In reply to: [346074810](https://github.com/microsoft/BuildXL/pull/1207#discussion_r346074810) [](ancestors = 346074810)

Reply:
Ah, okay, I thought bxl had two deployments, one for full framework, with the full framework binaries copied, and one netcore, with the netcore binaries copies (some of which will be the same, i.e. netstandard).